File: aoc_25/2/golang/main.go
package main

import (
	"fmt"
	"os"
	"possum/invalids"
	"strings"
)

func main() {
	inputBytes, err := os.ReadFile("input")
	if err != nil {
		panic(err)
	}
	input := string(inputBytes)
	stringIter := strings.SplitSeq(input, ",")
	total := 0
	for part := range stringIter {
		parts := strings.Split(part, "-")
		from := parts[0]
		to := parts[1]
		repeating := invalids.GetFromRange(from, to)
		fmt.Println(part, repeating)
		for _, r := range repeating {
			total += r
		}
	}
	fmt.Println("total:", total)
}

File: aoc_25/2/golang/main.py
with open("input") as wfile:
    parts = wfile.read().split(",")
    total = 0
    for part in parts:
        if not part:
            continue
        start, end = part.split("-")
        print(repr(part), repr(start), repr(end))
        for i in range(int(start), int(end) + 1):
            s = str(i)
            length = len(s)
            if length % 2 != 0:
                continue
            if s[length//2:] * 2 == s:
                total += i
                # print(i)
    print(total)
